### DISCUSSION TITLE
Parameter identifiability in race models

### DISCUSSION THREAD

**Discussion by jainraj** [lnccbrown/HSSM#591] [Q&A]
I currently use a 3-particle race model (different drift rates and different starting points per particle, single threshold, and non-decision time) to model a 3-choice experimental paradigm. I would like to know how identifiable the parameters of such a model are. I would like to know 
1. What parameter combinations could lead to identical output distributions? 
2. Consequently, what reparameterizations could be helpful to use?
3. Any reference(s) that have theoretical results - possibly something on what conditions to be imposed on the parameters to guarantee identifiability (I understand that behaviorally/biologically, those conditions may not hold true) 

One hunch for point number 1: increasing all the starting points by a small value (other parameters unchanged) would lead to a leftward shift in the RT distribution. Thus, I could get an identical distribution with a lower non-decision time and the changed starting points. (Please let me know if this sounds right). A possible solution would be to fix one of the starting points and analyze other starting points relative to this.

**Comment by frankmj** [comment]
Hi Raj, in absence of a known analytical solution to these questions you
can simply conduct a parameter recovery study by generating data from a
range of parameters and looking at the identifiability in the posterior
distributions (including the joint distributions in posterior pair plots
which will tell you how confusable one parameter is with another).

Often identifiability is also helped a lot if one puts constraints - eg.
you have multiple task conditions and assume that most parameters are
shared across those conditions but only one or a subset of them are
allowed to vary by condition (this is often theoretically justified). Then
when fitting this provides constraints that can help pin down some
parameters and improve the identifiability of others.
M

Michael J Frank, PhD | Edgar L. Marston Professor
Director, Carney Center for Computational Brain Science
<https://www.brown.edu/carney/ccbs>
Laboratory of Neural Computation and Cognition <https://www.lnccbrown.com/>

Brown University
website <http://ski.clps.brown.edu>

**Comment by AlexanderFengler** [comment]
Hi @jainraj, 

A few pointers on these race models, since I had looked into those at some point. 
In the `ssm-simulators` package (our simulation backend), you will generally find three versions of these race models. Setting aside some need for more elegant renaming: 

1. The standard version `race_3` etc.. This includes a bias (starting point) parameter for each accumulator (particle is the terminology used in the code)
2. A version that has `_no_bias` suffix, which reduced the number of bias parameters to one global bias (starting point)
3. A version that has the `_no_z` suffix, which simply fixes the bias parameter (I think to `0.5` have to double check)

Overall I recommend to use the `_no_z` version, since indeed, you can generally expect very high posterior correlation between the `bias` and the `a` (boundary) parameter. 

Best,
Alex

**Reply by jainraj** [reply]
Thank you, Alex. According to our hypothesis, we expect to see effects in the bias of one of the choices, hence the reason for fitting one starting per choice. Would it make sense to fix the threshold to some value - let's say a=2 and then fit the starting points?

**Reply by AlexanderFengler** [reply]
Hey @jainraj,

sorry I had missed this. 
The answer would have been 'yes' :). 

Did you make any headway on that?

**Reply by jainraj** [reply]
Thank you, Alex, for the confirmation :) Parameter identifiability with a fixed threshold does become better. I am currently looking at parameter identifiability with multiple constraints (Prof. Frank's suggestion) - empirically.
